app/views/articles/_article.html.erb

```diff
@@ -0,0 +1,31 @@
+<% @articles.each do |article| %>
+  <div class="card text-center">
+    <% if user_signed_in? %>
+      <div class="card-header">
+        Written by: <%= username(article.user.email) %>
+      </div>
+    <% else %>
+      <div class="card-header">
+        You need to sign up or login to see the author
+      </div>
+    <% end %>
```

Comment:
Remove logic from views:
you can put the whole if statement in a method called show_artilce under articles helpers
```ruby
def show_artilce(article) 
      if user_signed_in? %>
          render html: tag.div "Written by: ${username(article.user.email})"
     else
         render html: tag.div "You need to sign up or login to see the author"
     end
end
```
call show article method from your view,
Please do the same for all logics you have in your views